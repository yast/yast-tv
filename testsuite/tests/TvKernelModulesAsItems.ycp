{

    include "testsuite.ycp";

    map READ = $[
	"etc" : $[ "install_inf" : $[] ],
	"target" : $[ "ycp" : $[] ],
	"probe" : $[ "system" : []],
    ];

    TESTSUITE_INIT([READ], nil);
    import "Tv";

	define TEST_It_Now(string s_module) ``{
	DUMP("---------------");

	// kernel_modules_database cannot be nil (see ReadKernelModules)
	/*Tv::kernel_modules = nil;
	list items = Tv::KernelModulesAsItems(s_module);
	DUMP(items);*/


	Tv::kernel_modules = $[];
	list items = Tv::KernelModulesAsItems(s_module);
	DUMP(items);


	Tv::kernel_modules = $[
		"bttv" : $["module_description":"bttv - v4l driver module for bt848/878 based cards"],
		"zoran": $["module_description":"Zoran ZR36120 based framegrabber"]
	];
	items = Tv::KernelModulesAsItems(s_module);
	DUMP(items);

	}

    // testing with different values of selected_module
    TEST_It_Now(nil);

    TEST_It_Now("bttv");

	TEST_It_Now("zoran");
}
