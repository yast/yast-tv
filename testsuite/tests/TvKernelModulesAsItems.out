Read	.target.size "/etc/install.inf" nil
Read	.etc.install_inf.Sourcemounted nil
Read	.etc.install_inf.AutoYaST nil
Read	.etc.install_inf.Demo nil
Read	.etc.install_inf.Textmode nil
Read	.etc.install_inf.Manual nil
Read	.etc.install_inf.Console nil
Read	.etc.install_inf.Braille nil
Read	.etc.install_inf.VNC nil
Read	.etc.install_inf.Locale nil
Read	.etc.install_inf.Instmode nil
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp nil
Read	.probe.system []
Dump	---------------
Dump	[]
Dump	[`item (`id ("bttv"), "bttv: bttv - v4l driver module for bt848/878 based cards", false), `item (`id ("zoran"), "zoran: Zoran ZR36120 based framegrabber", false)]
Dump	---------------
Dump	[]
Dump	[`item (`id ("bttv"), "bttv: bttv - v4l driver module for bt848/878 based cards", true), `item (`id ("zoran"), "zoran: Zoran ZR36120 based framegrabber", false)]
Dump	---------------
Dump	[]
Dump	[`item (`id ("bttv"), "bttv: bttv - v4l driver module for bt848/878 based cards", false), `item (`id ("zoran"), "zoran: Zoran ZR36120 based framegrabber", true)]
